File: c-003/main.go
package main

func LargestPrimeFactor(num int) int {
	for i := 2; i < num; i++ {
		if num%i == 0 {
			return LargestPrimeFactor(num / i)
		}
	}
	return num
}

func main() {
	println(LargestPrimeFactor(600_851_475_143))
}

File: c-003/main.rs
fn largest_prime_factor(num: i64) -> i64 {
    for i in 2..(num-1) {
        if num % i == 0 {
            return largest_prime_factor(num / i);
        }
    }
    return num;
}

fn main() {
    println!("{}", largest_prime_factor(600_851_475_143));
}
